packages/opencensus-propagation-tracecontext/src/tracecontext-format.ts

```diff
@@ -16,18 +16,51 @@
 
 import {HeaderGetter, HeaderSetter, Propagation, SpanContext} from '@opencensus/core';
 import * as crypto from 'crypto';
-import {isValidOption, isValidSpanId, isValidTraceId, isValidVersion} from './validators';
+import {isValidSpanId, isValidTraceId, isValidVersion} from './validators';
 
-// Header names
+/** The traceparent header key. */
 export const TRACE_PARENT = 'traceparent';
+/** The tracestate header key. */
 export const TRACE_STATE = 'tracestate';
+/** The default trace options. This defaults to unsampled. */
 export const DEFAULT_OPTIONS = 0x0;
+/** Regular expression that represents a valid traceparent header. */
+const TRACE_PARENT_REGEX = /^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$/;
+
+/**
+ * Parses a traceparent header value into a SpanContext object, or null if the
+ * traceparent value is invalid.
+ */
+function traceParentToSpanContext(traceParent: string): SpanContext|null {
+  const match = traceParent.match(TRACE_PARENT_REGEX);
+  if (!match) return null;
+  const parts = traceParent.split('-');
+  const version = parts[0];
+  const traceId = parts[1];
+  const spanId = parts[2];
+  // tslint:disable-next-line:ban Needed to parse hexadecimal.
+  const options = parseInt(parts[3], 16);
+
+  if (!isValidVersion(version) || !isValidTraceId(traceId) ||
+      !isValidSpanId(spanId)) {
+    return null;
+  }
+  return {traceId, spanId, options};
+}
+
+/** Converts a headers type to a string. */
+function parseHeader(str: string|string[]|undefined): string|undefined {
+  if (Array.isArray(str)) {
```

Comment:
Optional: this could be a ternary: `return Array.isArray(str) ? str[0] : str;`

Reply:
done.